test.py

```diff
@@ -11,6 +11,46 @@ def g(s):
     assert f_g.functions == (g, f)  # functions exposed in execution order
 
 
+def test_compose_add_compose():
+    def f(s):
+        return s + 'f'
+    def g(s):
+        return s + 'g'
+    f_wrapped = compose(f)
+    g_wrapped = compose(g)
+    f_g_wrapped = f_wrapped + g_wrapped
+
+    assert f_g_wrapped('') == 'gf'  # Remember that it's f(g(...)) so g runs first
+    assert f_g_wrapped.functions == (g, f)  # functions exposed in execution order
+
+
+def test_compose_add_function():
+    def f(s):
+        return s + 'f'
+    def g(s):
+        return s + 'g'
+    f_wrapped = compose(f)
+    f_g_composed = f_wrapped + g
+    assert f_g_composed('') == 'gf'  # Remember that it's f(g(...)) so g runs first
+    assert f_g_composed.functions == (g, f)  # functions exposed in execution order
+
+
+def test_compose_add_neither_compose_nor_function():
+    def f(s):
+        return s + 'f'
+    f_wrapped = compose(f)
+    g = 'string'
+    
+    raised = False
+
+    try:
+        f_g_wrapped = f_wrapped + g
+    except ValueError as err:
+        raised = True
+    finally:
+        assert raised
+    
+
```

Comment:
Also, I think these tests only felt like they belonged in this spot of the file because the first two tests start out by re-testing everything that the first test tests. But once we clear that up, it becomes more obvious that these tests are not particularly related to the first test at all, and thus are just testing a distinct feature.

And in fact they're breaking up the flow of the tests, because the tests that immediately follow are still testing `compose` functionality, while thesenew tests are testing the layered-on addition of `+` behaving as an alias/shorthand for `compose`.